editor/state.js

```diff
@@ -10,17 +10,35 @@ import { keyBy, last, omit, without } from 'lodash';
 import { combineUndoableReducers } from 'utils/undoable-reducer';
 
 /**
- * Reducer returning editor blocks state, an combined reducer of keys byUid,
- * order, where blocks are parsed from current HTML markup.
+ * Undoable reducer returning the editor post state, including blocks parsed
+ * from current HTML markup.
+ *
+ * Handles the following state keys:
+ *  - post: an object describing the current post, in the format used by the WP
+ *  REST API
+ *  - blocksByUid: post content blocks keyed by UID
+ *  - blockOrder: list of block UIDs in order
  *
  * @param  {Object} state  Current state
  * @param  {Object} action Dispatched action
  * @return {Object}        Updated state
  */
-export const blocks = combineUndoableReducers( {
-	byUid( state = {}, action ) {
+export const editor = combineUndoableReducers( {
+	post( state = {}, action ) {
+		switch ( action.type ) {
+			case 'EDIT_POST':
+				return action.post || state;
+
+			case 'POST_UPDATE_REQUEST_SUCCESS':
+				return action.post;
+		}
+
+		return state;
+	},
+
+	blocksByUid( state = {}, action ) {
 		switch ( action.type ) {
-			case 'REPLACE_BLOCKS':
+			case 'EDIT_POST':
```

Comment:
Update - this PR is currently using the latest known version of the post object to determine if the currently edited post is new (has an ID) or not.  Also lots of interesting uses for this later:  with regular updates from the server, and conflict resolution where appropriate (a huge task, to be fair), we basically have collaborative editing.